book/src/programmability/witness-pattern.md

```diff
@@ -0,0 +1,277 @@
+# Pattern: Witness
+
+Almost every combination of [abilities](../move-basics/abilities-introduction.md) forms a pattern in
+Move. And one of the most important ones is the _Witness pattern_ - a type that with the
+[`drop`](./../move-basics/drop-ability.md) ability. It is used to instantiate generic types with a
+concrete type parameter by providing a _proof of ownership_ of this type.
+
+## Definition
+
+A Witness is a type that has the `drop` ability. It is not very useful on its own, within a single
+module, but rather serves as an initialization value for generic types. For example, if there is a
+generic type `Custom<T>`, where `T` is a type parameter, a witness may be used to instantiate it,
+while also ensuring that no other module can create a `Custom<T>` with the same `T`.
+
+```move
+{{#include ../../../packages/samples/sources/programmability/witness-pattern.move:definition}}
+```
+
+## Use Cases
+
```

Comment:
Another case similar to Proof of Ownership, is combining the witness with the hot potato pattern where you can return a witness in a function, and write a proof function that one of your functions had to have been called.